core/dbt/parser/util.py

```diff
@@ -124,96 +124,127 @@ def _get_node_column(cls, node, column_name):
+            cls.process_docs_for_node(manifest, current_project, node)
         return manifest
 
     @classmethod
-    def process_refs(cls, manifest, current_project):
-        for _, node in manifest.nodes.items():
-            target_model = None
-            target_model_name = None
-            target_model_package = None
-
-            for ref in node.refs:
-                if len(ref) == 1:
-                    target_model_name = ref[0]
-                elif len(ref) == 2:
-                    target_model_package, target_model_name = ref
-
-                target_model = cls.resolve_ref(
-                    manifest,
-                    target_model_name,
-                    target_model_package,
-                    current_project,
-                    node.get('package_name'))
-
-                if target_model is None or target_model is cls.DISABLED:
-                    # This may raise. Even if it doesn't, we don't want to add
-                    # this node to the graph b/c there is no destination node
-                    node.config['enabled'] = False
-                    dbt.utils.invalid_ref_fail_unless_test(
-                            node, target_model_name, target_model_package,
-                            disabled=(target_model is cls.DISABLED)
-                    )
-
-                    continue
-
-                target_model_id = target_model.get('unique_id')
-
-                node.depends_on['nodes'].append(target_model_id)
-                manifest.nodes[node['unique_id']] = node
+    def process_refs_for_node(cls, manifest, current_project, node):
+        """Given a manifest and a node in that manifest, process its refs"""
+        target_model = None
+        target_model_name = None
+        target_model_package = None
+
+        for ref in node.refs:
+            if len(ref) == 1:
+                target_model_name = ref[0]
+            elif len(ref) == 2:
+                target_model_package, target_model_name = ref
+
+            target_model = cls.resolve_ref(
+                manifest,
+                target_model_name,
+                target_model_package,
+                current_project,
+                node.get('package_name'))
+
+            if target_model is None or target_model is cls.DISABLED:
+                # This may raise. Even if it doesn't, we don't want to add
+                # this node to the graph b/c there is no destination node
+                node.config['enabled'] = False
+                dbt.utils.invalid_ref_fail_unless_test(
+                        node, target_model_name, target_model_package,
+                        disabled=(target_model is cls.DISABLED)
+                )
+
+                continue
+
+            target_model_id = target_model.get('unique_id')
+
+            node.depends_on['nodes'].append(target_model_id)
+            manifest.nodes[node['unique_id']] = node
 
+    @classmethod
+    def process_refs(cls, manifest, current_project):
```

Comment:
just want to raise this, though unsure if it's relevant in this particular part of the codebase: When we worked on a feature like this previously, dbt's RuntimeContext choked on ad-hoc compilation of analysis nodes that contained refs. The RuntimeContext expected to see the `ref` in the list of refs for the ParsedNode, but we skipped parsing (i think) so there was no ref! As a result, dbt served up an error message instructing us to hoist our refs......

Ignore if irrelevant, happy to say more about this if it proves to be something we'll need to address